src/components/NavBar/ChainSelector.tsx

```diff
@@ -69,9 +69,13 @@ export const ChainSelector = ({ leftAlign }: ChainSelectorProps) => {
   const theme = useTheme()
 
   const showTestnets = useAtomValue(showTestnetsAtom)
-  const chains = showTestnets
-    ? NETWORK_SELECTOR_CHAINS
-    : NETWORK_SELECTOR_CHAINS.filter((chain) => !TESTNET_CHAIN_IDS.has(chain))
+  const walletSupportsChain = useWalletSupportedChains()
+
+  const chains = useMemo(() => {
+    return (
+      showTestnets ? NETWORK_SELECTOR_CHAINS : NETWORK_SELECTOR_CHAINS.filter((chain) => !TESTNET_CHAIN_IDS.has(chain))
```

Comment:
nit: stick the ternary inside the filter function so it's easier to follow, ie 'return showTestnets ? true : TESTNET_CHAIN_IDS.has(chainId)'

Reply:
Could you also add unit tests around this logic?